python/mxnet/test_utils.py

```diff
@@ -116,7 +116,22 @@ def _validate_csr_generation_inputs(num_rows, num_cols, density,
                              % (density, num_rows, num_cols))
 
 
-def _get_uniform_dataset_csr(num_rows, num_cols, density=0.1, dtype=None):
+def shuffle_csr_column_indices(csr):
+    """Shuffle CSR column indices per row
+    This allows validation of unordered column indices, which is not a requirement
+    for a valid CSR matrix
+    """
+    row_count = len(csr.indptr) - 1
+    for i in range(row_count):
+        start_index = csr.indptr[i]
+        end_index = csr.indptr[i + 1]
+        sublist = np.array(csr.indices[start_index : end_index])
+        np.random.shuffle(sublist)
+        csr.indices[start_index : end_index] = sublist
+
+
+def _get_uniform_dataset_csr(num_rows, num_cols, density=0.1, dtype=None,
```

Comment:
I asked the same question to our customers, they said they need to ask the same question to their customers to know whether there's such use case. It's safe to say that so far the datasets they're using have each row with sorted indices. But in general CSR definition doesn't specify that indices have to be sorted. 
I'll post an update once I got more feedback from users.